x-pack/metricbeat/module/azure/monitor_service.go

```diff
@@ -11,134 +11,284 @@ import (
-	metricNamespaceClient  *insights.MetricNamespacesClient
-	resourceClient         *resources.Client
+	metricsClient          *armmonitor.MetricsClient
+	metricDefinitionClient *armmonitor.MetricDefinitionsClient
+	metricNamespaceClient  *armmonitor.MetricNamespacesClient
+	resourceClient         *armresources.Client
 	context                context.Context
 	log                    *logp.Logger
 }
 
 const (
 	metricNameLimit = 20
-	ApiVersion      = "2019-12-01"
+	ApiVersion      = "2021-04-01"
 )
 
 // NewService instantiates the Azure monitoring service
 func NewService(config Config) (*MonitorService, error) {
-	clientConfig := auth.NewClientCredentialsConfig(config.ClientId, config.ClientSecret, config.TenantId)
-	clientConfig.AADEndpoint = config.ActiveDirectoryEndpoint
-	clientConfig.Resource = config.ResourceManagerEndpoint
-	authorizer, err := clientConfig.Authorizer()
+	cloudServicesConfig := cloud.AzurePublic.Services
+
+	if config.ResourceManagerEndpoint != "" && config.ResourceManagerEndpoint != DefaultBaseURI {
+		cloudServicesConfig[cloud.ResourceManager] = cloud.ServiceConfiguration{
+			Audience: config.ResourceManagerEndpoint,
+			Endpoint: config.ResourceManagerEndpoint,
+		}
+	}
+
+	credential, err := azidentity.NewClientSecretCredential(config.TenantId, config.ClientId, config.ClientSecret,
+		&azidentity.ClientSecretCredentialOptions{
+			ClientOptions: policy.ClientOptions{
+				Cloud: cloud.Configuration{
+					Services:                     cloudServicesConfig,
+					ActiveDirectoryAuthorityHost: config.ActiveDirectoryEndpoint,
+				},
+			},
+		})
 	if err != nil {
-		return nil, err
-	}
-	metricsClient := insights.NewMetricsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
-	metricsDefinitionClient := insights.NewMetricDefinitionsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
-	resourceClient := resources.NewClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
-	metricNamespaceClient := insights.NewMetricNamespacesClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
-	metricsClient.Authorizer = authorizer
-	metricsDefinitionClient.Authorizer = authorizer
-	resourceClient.Authorizer = authorizer
-	metricNamespaceClient.Authorizer = authorizer
+		return nil, fmt.Errorf("couldn't create client credentials: %w", err)
+	}
+
+	metricsClient, err := armmonitor.NewMetricsClient(credential, &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: cloud.Configuration{
+				Services:                     cloudServicesConfig,
+				ActiveDirectoryAuthorityHost: config.ActiveDirectoryEndpoint,
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create metrics client: %w", err)
+	}
+
+	metricsDefinitionClient, err := armmonitor.NewMetricDefinitionsClient(credential, &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: cloud.Configuration{
+				Services:                     cloudServicesConfig,
```

Comment:
Nit: I see we're creating one `policy.ClientOptions` for each client. Is this required, or we can use the same `policy.ClientOptions` on multiple clients?